src/order/liminf_limsup.lean

```diff
@@ -678,8 +698,72 @@ end
   bliminf u f (λ x, p x ∨ q x) = bliminf u f p ⊓ bliminf u f q :=
 @blimsup_or_eq_sup αᵒᵈ β _ f p q u
 
+lemma sup_limsup [ne_bot f] (a : α) :
+  a ⊔ limsup u f = limsup (λ x, a ⊔ u x) f :=
+begin
+  simp only [limsup_eq_infi_supr, supr_sup_eq, sup_binfi_eq],
+  congr, ext s, congr, ext hs, congr,
+  exact (bsupr_const (nonempty_of_mem hs)).symm,
+end
+
+lemma inf_liminf [ne_bot f] (a : α) :
+  a ⊓ liminf u f = liminf (λ x, a ⊓ u x) f :=
+@sup_limsup αᵒᵈ β _ f _ _ _
+
+lemma sup_liminf (a : α) :
+  a ⊔ liminf u f = liminf (λ x, a ⊔ u x) f :=
+begin
+  simp only [liminf_eq_supr_infi],
+  rw [sup_comm, bsupr_sup (⟨univ, univ_mem⟩ : ∃ (i : set β), i ∈ f)],
+  simp_rw [binfi_sup_eq, @sup_comm _ _ a],
+end
+
+lemma inf_limsup (a : α) :
+  a ⊓ limsup u f = limsup (λ x, a ⊓ u x) f :=
+@sup_liminf αᵒᵈ β _ f _ _
+
 end complete_distrib_lattice
 
+section complete_boolean_algebra
+
+variables [complete_boolean_algebra α] (f : filter β) (u : β → α)
+
+lemma limsup_compl :
+  (limsup u f)ᶜ = liminf (compl ∘ u) f :=
+by simp only [limsup_eq_infi_supr, liminf_eq_supr_infi, compl_infi, compl_supr]
+
+lemma liminf_compl :
+  (liminf u f)ᶜ = limsup (compl ∘ u) f :=
+by simp only [limsup_eq_infi_supr, liminf_eq_supr_infi, compl_infi, compl_supr]
+
+lemma limsup_sdiff (a : α) :
+  (limsup u f) \ a = limsup (λ b, (u b) \ a) f :=
+begin
+  simp only [limsup_eq_infi_supr, sdiff_eq],
+  rw binfi_inf (⟨univ, univ_mem⟩ : ∃ (i : set β), i ∈ f),
+  simp_rw [inf_comm, inf_bsupr_eq, inf_comm],
+end
+
+lemma liminf_sdiff [ne_bot f] (a : α) :
+  (liminf u f) \ a = liminf (λ b, (u b) \ a) f :=
+by simp only [sdiff_eq, @inf_comm _ _ _ aᶜ, inf_liminf]
+
+lemma sdiff_limsup [ne_bot f] (a : α) :
+  a \ (limsup u f) = liminf (λ b, a \ (u b)) f :=
+begin
+  rw ← compl_inj_iff,
+  simp only [sdiff_eq, liminf_compl, (∘), compl_inf, compl_compl, sup_limsup],
+end
+
+lemma sdiff_liminf (a : α) :
+  a \ (liminf u f) = limsup (λ b, a \ (u b)) f :=
```

Comment:
```suggestion
  a \ liminf u f = limsup (λ b, a \ u b) f :=
```
and elsewhere

Reply:
OK done, except I have kept the brackets in expressions of the form `(...) \ y` since I think they are a worthwhile aid to readability.